src/Raven.Server/Documents/DocumentsTransaction.cs

```diff
@@ -17,6 +17,8 @@ public class DocumentsTransaction : RavenTransaction
 
         private List<DocumentChange> _documentNotifications;
 
+        private List<CounterChange> _counterNotifications;
```

Comment:
We might want to have a pool of _documentNotifications and _counterNotifications